File: lib/src/modules/profile/dtos/update_address_request.dart
class UpdateAddressRequest {
  final int addressId;
  final String recipientName;
  final String phone;
  final String provinceOrCity;
  final String district;
  final String ward;
  final String addressLine;

  UpdateAddressRequest({
    required this.addressId,
    required this.recipientName,
    required this.phone,
    required this.provinceOrCity,
    required this.district,
    required this.ward,
    required this.addressLine,
  });

  Map<String, dynamic> toJson() {
    return {
      'addressId': addressId,
      'recipientName': recipientName,
      'phone': phone,
      'provinceOrCity': provinceOrCity,
      'district': district,
      'ward': ward,
      'addressLine': addressLine
    };
  }
}

File: src/main/java/com/example/sneakerx/dtos/user/UpdateAddressRequest.java
package com.example.sneakerx.dtos.user;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class UpdateAddressRequest {
    private Integer addressId;
    private String recipientName;
    private String phone;
    private String provinceOrCity;
    private String district;
    private String ward;
    private String addressLine;
    private Boolean isDefault;
}
